tracing/tracing.go

```diff
@@ -2,78 +2,192 @@ package tracing
-	cfg, err := jaegercfg.FromEnv()
+func (cfg TracingConfig) initJaegerTracerProvider(serviceName string) (*tracesdk.TracerProvider, error) {
+	// Read environment variables to configure Jaeger
+	var ep jaegerotel.EndpointOption
+	// Create the jaeger exporter: address can be either agent address (host:port) or collector URL
+	if cfg.host != "" {
+		ep = jaegerotel.WithAgentEndpoint(jaegerotel.WithAgentHost(cfg.host), jaegerotel.WithAgentPort(strconv.Itoa(cfg.port)))
+	} else {
+		ep = jaegerotel.WithCollectorEndpoint(jaegerotel.WithEndpoint(cfg.collectorURL))
+	}
+	exp, err := jaegerotel.New(ep)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
+		return nil, err
 	}
 
-	if cfg.Sampler.SamplingServerURL == "" && cfg.Reporter.LocalAgentHostPort == "" && cfg.Reporter.CollectorEndpoint == "" {
-		return nil, ErrBlankTraceConfiguration
+	res, err := NewResource(serviceName, cfg.customAttributes)
+	if err != nil {
+		return nil, err
 	}
 
-	return installJaeger(serviceName, cfg, options...)
+	// Configure sampling strategy
+	sampler := tracesdk.AlwaysSample()
+	if cfg.samplerType == "const" || cfg.samplerType == "probabilistic" {
+		sampler = tracesdk.TraceIDRatioBased(cfg.samplerParam)
+	} else if cfg.samplerType == "remote" {
+		sampler = jaegerremote.New(serviceName, jaegerremote.WithSamplingServerURL(cfg.samplerRemoteURL),
+			jaegerremote.WithInitialSampler(tracesdk.TraceIDRatioBased(cfg.samplerParam)))
+	} else if cfg.samplerType != "" {
+		return nil, errors.Errorf("unknown sampler type %q", cfg.samplerType)
+	}
+	tp := tracesdk.NewTracerProvider(
+		tracesdk.WithBatcher(exp),
+		tracesdk.WithResource(res),
+		tracesdk.WithSampler(sampler),
+	)
+
+	otel.SetTracerProvider(tp)
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(
+		propagation.TraceContext{},
+		propagation.Baggage{},
+	))
+
+	// OpenTracing <=> OpenTelemetry bridge
+	otTracer, _ := bridge.NewTracerPair(tp.Tracer(serviceName))
+	opentracing.SetGlobalTracer(otTracer)
+
+	return tp, nil
 }
 
 // ExtractTraceID extracts the trace id, if any from the context.
 func ExtractTraceID(ctx context.Context) (string, bool) {
-	sp := opentracing.SpanFromContext(ctx)
-	if sp == nil {
-		return "", false
-	}
-	sctx, ok := sp.Context().(jaeger.SpanContext)
-	if !ok {
-		return "", false
-	}
-
-	return sctx.TraceID().String(), true
+	traceId := trace.SpanContextFromContext(ctx).TraceID().String()
+	return traceId, traceId != ""
 }
 
-// ExtractSampledTraceID works like ExtractTraceID but the returned bool is only
-// true if the returned trace id is sampled.
 func ExtractSampledTraceID(ctx context.Context) (string, bool) {
-	sp := opentracing.SpanFromContext(ctx)
-	if sp == nil {
-		return "", false
+	traceID := trace.SpanContextFromContext(ctx).TraceID().String()
+	if traceID != "" {
+		return traceID, trace.SpanContextFromContext(ctx).IsSampled()
 	}
-	sctx, ok := sp.Context().(jaeger.SpanContext)
```

Comment:
Here I am not sure whether a cast of spancontext to jaeger.SpanContext is needed anymore